.buildbot.sh

```diff
@@ -6,12 +6,17 @@ export CARGO_HOME="`pwd`/.cargo"
 export RUSTUP_HOME="`pwd`/.rustup"
 
 curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs > rustup.sh
-# Install no toolchain initially to ensure toolchain rollback.
-sh rustup.sh --default-host x86_64-unknown-linux-gnu --default-toolchain none -y --no-modify-path
-
+sh rustup.sh --default-host x86_64-unknown-linux-gnu --default-toolchain nightly -y --no-modify-path
 export PATH=`pwd`/.cargo/bin/:$PATH
-rustup install nightly
+rustup toolchain install nightly --allow-downgrade --profile minimal
```

Comment:
What's the reasoning behind `--allow-downgrade` here?

I think you should be able to install the *default* profile *without* `--allow-downgrade` (for the build) and then later run rustup again *with* `--allow-downgrade` (to get any potentially missing rustup).

Reply:
I find rustup deeply unintuitive, so "I don't know" is part of the answer. Let me try on rboehm (long story) a variation to see if this might work.

Reply:
Admittedly, I had not heard of `--allow-downgrade` before, so maybe it doesn't behave as I expect.